airflow/providers/google/cloud/hooks/dataproc.py

```diff
@@ -218,11 +217,14 @@ def get_cluster_client(self, location: Optional[str] = None) -> ClusterControlle
             credentials=self._get_credentials(), client_info=self.client_info, client_options=client_options
         )
 
-    @cached_property
-    def get_template_client(self) -> WorkflowTemplateServiceClient:
+    def get_template_client(self, location: Optional[str] = None) -> WorkflowTemplateServiceClient:
         """Returns WorkflowTemplateServiceClient."""
+        client_options = None
+        if location and location != 'global':
```

Comment:
Do we need the case for `global`?

Reply:
If template is in global namespace, region (location) parameter needed to be set to 'global' as required for template path [1], [2].

[1]: https://github.com/apache/airflow/blob/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5/airflow/providers/google/cloud/hooks/dataproc.py#L645
[2]: https://github.com/googleapis/python-dataproc/blob/7ec92b71be9c1d0d305421bb1b1dce5d92377bba/google/cloud/dataproc_v1beta2/services/workflow_template_service/client.py#L155

Reply:
@otourzan in first link users have to provide the value for `location` so it's up to them to provide proper configuration. Or do I miss something?

Reply:
Btw. I see that you are Google related, are you on Airflow slack? My colleague @michalslowikowski00 is refactoring Dataproc operators to use dataproc python library 2.0 and having someone to consult would be lovely 🚀

Reply:
Yes, user should provide proper configuration. If they pass 'global' as a valid location, the endpoint needs to be global one (without region prefix: 'dataproc.googleapis.com') as there is endpoint like 'global-dataproc.googleapis.com'. Not sure if I got your point here.

I joined the Airflow slack and will be happy to help.